azDevOps/azure/templates/v2/steps/test-static-code-sonar.yml

```diff
@@ -8,12 +8,13 @@
 parameters:
   working_directory: ""
   target_container: ""
+  sonar_command: "sonar-scanner"
```

Comment:
Awesome. The default ensures that this is backwards compatible. If we do want to create a new version of the step, it would be great to also pass through the env vars as parameters, like in:

eg. `env_vars: {}`

The pass them through to the step iterating through:
```
    env:
      ${{ each var in parameters.env_vars }}:
        ${{ var.key }}: ${{ var.value }}
```

This ensures that the step can be consumed by any job regardless of implementation. @dnitsch FYI in light of our call.

Reply:
Aren't Env vars passed automatically?

Reply:
Also I'm not 100% sure passing all env vars into a container are a good idea, you might override something explicitly set into the container?

Reply:
Any env vars that are in the pipeline context aren't lost, this is just performing a merge between variable objects to ensure that the user if needed can pass through additional variables that may not be set.